### PR TITLE
Feature/add user to group

### PR DESCRIPTION
**Pull request by mwnicks**
### What

Added endpoint for adding a user to a group

### How to review

Ensure all the unit and feature test are passing
Send a POST request to /v1/groups/role-admin/members, with body:
{
    "user_id": "ca83ca4c-0495-4254-a131-57ada4b1fc70"
}

You should receive a 200 response with body that looks like:
{
    "name": "role-admin",
    "description": "The publishing admins",
    "precedence": 1,
    "created": "2021-07-08T16:59:24.---Z"
    "members": [
        {
            "forename": "Matt",
            "lastname": "Nicks",
            "email": "matt.nicks@ons.gov.uk",
            "groups": [],
            "status": "CONFIRMED",
            "active": true,
            "id": "ca83ca4c-0495-4254-a131-57ada4b1fc70",
            "status_notes": "reactivating user"
        }
    ]
}

### Who can review

Anyone but me

### CODE REVIEW

File: api/groups.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import (
+	"context"
+	"encoding/json"
+	"github.com/ONSdigital/dp-identity-api/models"
+	"github.com/gorilla/mux"
+	"io/ioutil"
+	"net/http"
+)
+
+//ListUsersHandler lists the users in the user pool
```
Comment (morses-code) [inline api/groups.go]: Godoc needs updating here

**Review by morses-code — COMMENTED** [review]
Just a godoc comment from me, LGTM otherwise but another set of eyes would be good.

File: swagger.yaml [hunk 2]
```diff
@@ -365,6 +401,27 @@ definitions:
...
+        description: "The group name"
+        type: string
+      description:
+        description: "A description of the group"
+        type: string
+      precedence:
+        description: "The priority of the group"
+        type: integer
+      created:
+        description: "The the date the group was created"
```
Comment (HeJones) [inline swagger.yaml]: Just a small typo

File: swagger.yaml [hunk 3]
```diff
@@ -350,6 +348,44 @@ paths:
...
+      consumes:
+        - application/json
+      produces:
+        - "application/json"
+      parameters:
+        - in: path
+          name: name
+          type: string
+          required: true
+          description: the groups name
```
Comment (HeJones) [inline swagger.yaml]: should this be group's?

**Review by HeJones — COMMENTED** [review]
Just a couple of small comments on the swagger spec

**Review by HeJones — APPROVED** [review]
Looks good to me, and the endpoint gives the expected behaviour in Postman

**Pull request merged**